File: program_control/sources/nesting_restrict.2.c
/*
* @@name:	nesting_restrict.2
* @@type:	C
* @@operation:	compile
* @@expect:	unspecified
* @@version:	pre_omp_3.0
*/
void work(int i, int j) {}
void work1(int i, int n)
{
  int j;
/* incorrect nesting of loop regions */
  #pragma omp for
    for (j=0; j<n; j++)
      work(i, j);
}

void wrong2(int n)
{
  #pragma omp parallel default(shared)
  {
    int i;
    #pragma omp for
      for (i=0; i<n; i++)
         work1(i, n);
  }
}

File: program_control/sources/nesting_restrict.2.f
! @@name:	nesting_restrict.2
! @@type:	F-fixed
! @@operation:	compile
! @@expect:	unspecified
! @@version:	pre_omp_3.0
       SUBROUTINE WORK1(I,N)
       INTEGER I, N
       INTEGER J
!$OMP   DO      ! incorrect nesting of loop regions
        DO J = 1, N
          CALL WORK(I,J)
        END DO
       END SUBROUTINE WORK1
       SUBROUTINE WRONG2(N)
       INTEGER N
       INTEGER I
!$OMP   PARALLEL DEFAULT(SHARED)
!$OMP     DO
          DO I = 1, N
            CALL WORK1(I,N)
          END DO
!$OMP   END PARALLEL
       END SUBROUTINE WRONG2
